### ISSUE TITLE
Adaptive per-expert learning rate scheduling

### ISSUE THREAD

**Issue by kyo219** [kyo219/LightGBM-MoE#14]
## Problem

All experts share the same learning rate throughout training. In practice, some experts converge faster than others — converged experts overtrain while lagging experts undershoot.

## Proposed Solution

Track per-expert loss trends and adjust learning rate multipliers:

```cpp
struct ExpertScheduler {
    double loss_history[window_size];
    double lr_multiplier;  // range: [0.5, 2.0]
};
```

- If expert loss is decreasing steadily → reduce LR (fine-tuning phase)
- If expert loss is stagnating → increase LR (escape plateau)
- Combine with existing temperature annealing (`mixture_gbdt.cpp:1121-1129`)

## Expected Impact

5-10% faster convergence, better final model quality.

## Files

- `src/boosting/mixture_gbdt.cpp` (lines 1741-1833, M-step)
- `src/boosting/mixture_gbdt.h` (new member variables)
